src/openrct2/interface/InteractiveConsole.cpp

```diff
@@ -1924,6 +1924,67 @@ static int32_t ConsoleCommandProfilerStop(
     return 0;
 }
 
+static int32_t ConsoleCommandGoToEntity([[maybe_unused]] InteractiveConsole& console, [[maybe_unused]] const arguments_t& argv)
+{
+    if (argv.size() < 1)
+    {
+        console.WriteLineWarning("Too few arguments");
+        static_assert(EnumValue(EntityType::Count) == 13, "EntityType::Count changed, update console command!");
+        console.WriteLine("get_entity <entity ID> [entity type]");
+        console.WriteLine("entity type is one of:");
+        console.WriteLine("    0 (EntityType::Vehicle)");
+        console.WriteLine("    1 (EntityType::Guest)");
+        console.WriteLine("    2 (EntityType::Staff)");
+        console.WriteLine("    3 (EntityType::Litter)");
+        console.WriteLine("    4 (EntityType::SteamParticle)");
+        console.WriteLine("    5 (EntityType::MoneyEffect)");
+        console.WriteLine("    6 (EntityType::CrashedVehicleParticle)");
+        console.WriteLine("    7 (EntityType::ExplosionCloud)");
+        console.WriteLine("    8 (EntityType::CrashSplash)");
+        console.WriteLine("    9 (EntityType::ExplosionFlare)");
+        console.WriteLine("   10 (EntityType::JumpingFountain)");
+        console.WriteLine("   11 (EntityType::Balloon)");
+        console.WriteLine("   12 (EntityType::Duck)");
+        return 1;
+    }
+
+    EntityType entityType = EntityType::Null;
+    auto entityNumber = atoi(argv[0].c_str());
+    if (entityNumber > MAX_ENTITIES)
+    {
+        console.WriteFormatLine("Entity number out of bounds. Maximum: %d", MAX_ENTITIES);
+        return 1;
+    }
+    if (argv.size() >= 2)
+    {
+        entityType = static_cast<EntityType>(atoi(argv[1].c_str()));
+        if (EnumValue(entityType) >= EnumValue(EntityType::Count) && entityType != EntityType::Null)
+        {
+            console.WriteFormatLine(
+                "Entity type value out of bounds. Minimum: 0, maximum: %d", EnumValue(EntityType::Count) - 1);
+            return 1;
+        }
+    }
+    // This function requires manual updating for new entity types
+    EntityBase* entity = GetEntityByIndex(entityNumber, entityType);
```

Comment:
There is no need for the new `GetEntityByIndex` function.  It looks like you intend for it to check that the entity returned is of the desired type, but it doesn't do this. It only checks that the entity id is in bounds and then casts whatever the entity is into the requested type.

`EntityBase.EntityType` will tell you the type of the entity so you only need to compare it against the command parameter.

Reply:
Yeah the whole idea of the templated GetEntity is to check the type this is just needlessly undoing that. Just do:

```
auto* entity = TryGetEntity(entityId);
if (entity == nullptr || entity->EntityType != entityType)
{
   ... L1973
}
```